src/pretix/control/forms/event.py

```diff
@@ -1428,9 +1430,22 @@ class CountriesAndEU(CachedCountries):
     cache_subkey = 'with_any_or_eu'
 
 
+class CountriesAndEUAndStates(CountriesAndEU):
+    cache_subkey = 'with_any_or_eu_or_states'
+
+    def __iter__(self):
+        for country_code, country_name in super().__iter__():
+            yield country_code, country_name
+            if country_code in COUNTRIES_WITH_STATE_IN_ADDRESS:
+                types, form = COUNTRIES_WITH_STATE_IN_ADDRESS[country_code]
+                yield from sorted(((state.code, country_name + " - " + state.name)
```

Comment:
Not sure we can do much about it, but one thing I stumpled upon: in german translations there is a „USA – Sonstige kleine Inseln“ (United States Minor Outlying Islands) and „Vereinigte Staaten von Amerika“ with each state being „Vereinigte Staaten von Amerika – Alabama“. The thing I stumbled upon is the dash „-“ – with this „USA“ looks like a translation error. On the other hand, if it weren't called „USA – …“ it could be mistaken for being a subdivision of „Vereinigte Staaten von Amerika“ and by that one could assume „Vereinigte Staaten von Amerika“ would apply to the minor outlying islands as well – which it doesn’t as they do not have the same country-code.

TLDR; I played around with different separators, but dash just feels and looks the best. So we just live with „USA – Sonstige kleine Inseln“ looking like a subdivision, but being a „country“ of itself.